subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/swift/SwiftCompiler.java

```diff
@@ -111,7 +127,14 @@ public void execute(BuildOperationQueue<CommandLineToolInvocation> buildQueue) {
                     genericArgs.add("-emit-module-path");
                     genericArgs.add(spec.getModuleFile().getAbsolutePath());
                 }
+
                 genericArgs.add("-v");
+
+                if (canSafelyCompileIncrementally) {
```

Comment:
So `canSafelyCompileIncrementally` does not seem to take removed files into account.  But there is a test that checks that a removed file recompiles everything, so this apparently works.  Is this because swiftc itself notices that a file was removed and recompiles everything?

Reply:
Yep, https://github.com/apple/swift/blob/master/lib/Driver/Driver.cpp#L487-L489

I referenced this so much, I'll add this to the javadoc.